Sources/xcodeproj/Objects/Project/PBXProj.swift

```diff
@@ -218,7 +218,7 @@ extension PBXProj: Writable {
 
     public func write(path: Path, override: Bool, outputSettings: PBXOutputSettings) throws {
         let encoder = PBXProjEncoder(outputSettings: outputSettings)
-        let output = try encoder.encode(proj: self)
+        let output = encoder.encode(proj: self)
```

Comment:
For encoding / decoding throwing methods are expected I would have thought?

Reply:
Since the function is still a `throws` function it makes sense to keep `try`

Reply:
The try was removed because the throwing write function was removed - I'm thinking we need to revisit that to ensure it does still throw.

Reply:
We are not exactly following the encoding/decoding approach that Apple follows with the decoders here. The custom decoder that we are using here just writes the Xcode-plist representation to a string that we end up writing to the disk. The only scenario that I can think of in which the encoder could error is if users refer to objects that don't belong to any project. 

To prevent that I'll make the API safer in a follow-up PR.

Is there any other scenario that you think I might be missing here?